src/components/MagicCodeInput.tsx

```diff
@@ -226,110 +221,108 @@ function MagicCodeInput(props) {
         // edited and the number of digits added by the user.
         const numbersArr = addedValue
             .trim()
             .split('')
-            .slice(0, props.maxLength - editIndex);
-        const updatedFocusedIndex = Math.min(editIndex + (numbersArr.length - 1) + 1, props.maxLength - 1);
+            .slice(0, maxLength - editIndex);
+        const updatedFocusedIndex = Math.min(editIndex + (numbersArr.length - 1) + 1, maxLength - 1);
 
-        let numbers = decomposeString(props.value, props.maxLength);
-        numbers = [...numbers.slice(0, editIndex), ...numbersArr, ...numbers.slice(numbersArr.length + editIndex, props.maxLength)];
+        let numbers = decomposeString(value, maxLength);
+        numbers = [...numbers.slice(0, editIndex), ...numbersArr, ...numbers.slice(numbersArr.length + editIndex, maxLength)];
 
         setInputAndIndex(updatedFocusedIndex);
 
         const finalInput = composeToString(numbers);
-        props.onChangeText(finalInput);
+        onChangeTextProp(finalInput);
     };
 
     /**
      * Handles logic related to certain key presses.
      *
      * NOTE: when using Android Emulator, this can only be tested using
      * hardware keyboard inputs.
-     *
-     * @param {Object} event
      */
-    const onKeyPress = ({nativeEvent: {key: keyValue}}) => {
+    const onKeyPress = (event: Partial<NativeSyntheticEvent<TextInputKeyPressEventData>>) => {
+        const keyValue = event?.nativeEvent?.key;
         if (keyValue === 'Backspace' || keyValue === '<') {
-            let numbers = decomposeString(props.value, props.maxLength);
+            let numbers = decomposeString(value, maxLength);
 
             // If keyboard is disabled and no input is focused we need to remove
             // the last entered digit and focus on the correct input
-            if (props.isDisableKeyboard && focusedIndex === undefined) {
+            if (isDisableKeyboard && focusedIndex === undefined) {
                 const indexBeforeLastEditIndex = editIndex === 0 ? editIndex : editIndex - 1;
 
                 const indexToFocus = numbers[editIndex] === CONST.MAGIC_CODE_EMPTY_CHAR ? indexBeforeLastEditIndex : editIndex;
-                inputRefs.current[indexToFocus].focus();
-                props.onChangeText(props.value.substring(0, indexToFocus));
+                const formElement = inputRefs.current as HTMLFormElement | null;
+                (formElement?.[indexToFocus] as HTMLInputElement).focus();
+                onChangeTextProp(value.substring(0, indexToFocus));
 
                 return;
             }
 
             // If the currently focused index already has a value, it will delete
             // that value but maintain the focus on the same input.
-            if (numbers[focusedIndex] !== CONST.MAGIC_CODE_EMPTY_CHAR) {
+            if (focusedIndex !== undefined && numbers?.[focusedIndex] !== CONST.MAGIC_CODE_EMPTY_CHAR) {
```

Comment:
Could you check that it works same way as before in case of `focusedIndex` is `undefined`

Reply:
I checked, works the same 😄